docs/examples/image_quality/structural_similarity_index.jl

```diff
@@ -0,0 +1,82 @@
+# ---
+# cover: assets/ssim.png
+# title: Structural Similarity Index
+# ---
+
+# When comparing images, the **Mean Squared Error** (or MSE), though straightforward to calculate,
+# may not be a very good indicator of their *perceived* similarity.
+
+# The **Structural Similarity Index** (or SSIM) aims to address this shortcoming by taking texture
+# into account, and assigning a higher score to images that may *appear* similar.
+
+
+using Images, TestImages
+using Random
+
+img_orig = Gray{Float64}.(testimage("cameraman"))
+
+# We use a grayscale image out of the `TestImages` package, which provides a
+# standard suite of test images.
+
+assess_ssim(img_orig, img_orig)
+
+# The `assess_ssim` function, which takes two images as inputs and returns their
+# structural similarity index, is the simplest way to calculate the SSIM of two images.
+
+# An SSIM score of `1.00` indicates perfect structural similarity, as is expected
+# out of identical images.
+
+noise = ones(size(img_orig)) .* 0.2 .* (maximum(img_orig) - minimum(img_orig))
+img_const = img_orig + noise
+
+mask = rand(Float64, size(img_orig)) .< 0.5
+noise[mask] = noise[mask] .* -1
+img_noise = img_orig + noise
+
+mosaicview(img_const, img_noise; nrow=1)
+save("assets/ssim.png", img_noise) #src
+# Here, we create two variations of the original image: `image_const` on the left has the intensity of
+# all its pixels increased by `0.2` times the intensity range, while `image_noise` on the right has the
+# intensity of some of its pixels increased, and that of the others decreased by the same
+# amount. The two images look quite different visually.
+
+function mse(A::AbstractArray, B::AbstractArray)
+    if size(A) != size(B)
+        err = ArgumentError("images should be the same size, instead they are $(size(A))-$(size(B))")
+        throw(err)
+    end
+    D = (abs.(A-B)).^2
+    return sum(D)/length(D)
+end
+
+println(mse(img_orig, img_const))
+println(mse(img_orig, img_noise))
```

Comment:
niche picking:

```suggestion
assess_psnr(img_orig, img_const), assess_psnr(img_orig, img_noise)
```

This line can be moved downwards to make it more concise.